File: swift/DesignPattern.playground/Pages/SimpleFactoryPattern.xcplaygroundpage/Sources/Beverage.swift
import Foundation

public protocol Beverage {
    func addSuger(level: Int)
    func addIce(level: Int)
    func shake()
    func packageUp()
}

public extension Beverage {
    func addSuger(level: Int) {
        print("[\(self)] addSuger \(level)")
    }
    
    func addIce(level: Int) {
        print("[\(self)] addIce \(level)")
    }
    
    func shake() {
        print("[\(self)] shake")
    }
    
    func packageUp() {
        print("[\(self)] packageUp")
    }
}

File: kotlin/DesignPattern/src/main/kotlin/simplefactorypattern/source/Beverage.kt
package simplefactorypattern.source

interface Beverage {
    fun addSuger(level: Int) {
        println("[$this] addSuger $level")
    }

    fun addIce(level: Int) {
        println("[$this] addIce $level")
    }

    fun shake() {
        println("[$this] shake")
    }

    fun packageUp() {
        println("[$this] packageUp")
    }
}